tests/DuplexResourceStreamIntegrationTest.php

```diff
@@ -17,40 +17,38 @@ class DuplexResourceStreamIntegrationTest extends TestCase
 {
     public function loopProvider()
     {
-        return array(
-            array(
-                function() {
-                    return true;
-                },
-                function () {
-                    return new StreamSelectLoop();
-                }
-            ),
-            array(
-                function () {
-                    return function_exists('event_base_new');
-                },
-                function () {
-                    return class_exists('React\EventLoop\ExtLibeventLoop') ? new ExtLibeventLoop() : new LibEventLoop();
-                }
-            ),
-            array(
-                function () {
-                    return class_exists('libev\EventLoop');
-                },
-                function () {
-                    return class_exists('React\EventLoop\ExtLibevLoop') ? new ExtLibevLoop() : new LibEvLoop();
-                }
-            ),
-            array(
-                function () {
-                    return class_exists('EventBase') && class_exists('React\EventLoop\ExtEventLoop');
-                },
-                function () {
-                    return new ExtEventLoop();
-                }
-            )
-        );
+        yield [
+            function() {
+                return true;
+            },
+            function () {
+                return new StreamSelectLoop();
+            }
+        ];
+        yield [
+            function () {
+                return function_exists('event_base_new');
+            },
+            function () {
+                return class_exists('React\EventLoop\ExtLibeventLoop') ? new ExtLibeventLoop() : new LibEventLoop();
+            }
```

Comment:
@clue dropped the deprecated `ExtLibeventLoop` in https://github.com/reactphp/event-loop/pull/273 for EventLoop v3, so I think we don't have to test this here anymore:

```suggestion
            function () {
                return new LibEventLoop();
            }
```

Reply:
`LibEventLoop` doesn't exist.

Reply:
Dropping this one